packages/strapi-plugin-content-type-builder/admin/src/containers/HomePage/index.js

```diff
@@ -1,154 +1,144 @@
-  constructor(props) {
-    super(props);
+import ModelForm from '../ModelForm';
 
-    this.popUpHeaderNavLinks = [
-      { name: 'baseSettings', message: 'content-type-builder.popUpForm.navContainer.base', nameToReplace: 'advancedSettings' },
-      { name: 'advancedSettings', message: 'content-type-builder.popUpForm.navContainer.advanced', nameToReplace: 'baseSettings' },
-    ];
-  }
+import styles from './styles.scss';
 
-  handleButtonClick = () => {
-    if (storeData.getIsModelTemporary()) {
-      strapi.notification.info('content-type-builder.notification.info.contentType.creating.notSaved');
+class HomePage extends React.Component {
+  // eslint-disable-line react/prefer-stateless-function
+  handleClick = () => {
+    const {
+      canOpenModal,
+      history: { push },
+    } = this.props;
+    const { emitEvent } = this.context;
+
+    if (canOpenModal) {
+      emitEvent('willCreateContentType');
+      push({
+        search: 'modalType=model&settingType=base&actionType=create',
+      });
     } else {
-      // Send event.
-      this.context.emitEvent('willCreateContentType');
-      // Open CT modal.
-      this.toggleModal();
+      strapi.notification.info(`${pluginId}.notification.info.contentType.creating.notSaved`);
     }
-  }
-
-  handleDelete = (contentTypeName) => {
-    this.props.deleteContentType(contentTypeName, this.context);
-  }
-
-  toggleModal = () => {
-    const locationHash = this.props.location.hash ? '' : '#create::contentType::baseSettings';
-    router.push(`/plugins/content-type-builder/${locationHash}`);
-  }
-
-  renderTableListComponent = () => {
-    const availableNumber = size(this.props.models);
-    const title = availableNumber > 1 ? 'content-type-builder.table.contentType.title.plural'
-      : 'content-type-builder.table.contentType.title.singular';
-    return (
-      <TableList
-        availableNumber={availableNumber}
-        title={title}
-        buttonLabel="content-type-builder.button.contentType.add"
-        onButtonClick={this.handleButtonClick}
-        onHandleDelete={this.handleDelete}
-        rowItems={this.props.models}
-      />
-    );
-  }
+  };
 
   render() {
-    const component = size(this.props.models) === 0 ?
-      <EmptyContentTypeView handleButtonClick={this.toggleModal} />
-      : this.renderTableListComponent();
+    const {
+      cancelNewContentType,
+      canOpenModal,
+      connections,
+      createTempContentType,
+      deleteModel,
+      deleteTemporaryModel,
+      history: { push },
+      location: { pathname, search },
+      models,
+      modifiedData,
+      newContentType,
+      onChangeNewContentTypeMainInfos,
+    } = this.props;
+    const availableNumber = models.length;
+    const title =
+      availableNumber > 1
+        ? `${pluginId}.table.contentType.title.plural`
+        : `${pluginId}.table.contentType.title.singular`;
```

Comment:
Note: as there is no difference here `${pluginId}.table.contentType.title`
we can do something like that

```
const title = `${pluginId}.table.contentType.title.${availableNumber > 1 ? 'plural': 'singular'}`;
```